pdf/internal/e2etest/allobjects_test.go

```diff
@@ -0,0 +1,106 @@
+/*
+ * This file is subject to the terms and conditions defined in
+ * file 'LICENSE.md', which is part of this source code package.
+ */
+
+package e2etest
+
+import (
+	"errors"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/unidoc/unidoc/pdf/core"
+	"github.com/unidoc/unidoc/pdf/model"
+)
+
+// The allobjects test probes through all objects in PDF files of a specified corpus, decoding streams.
+// Set environment variables:
+//		UNIDOC_E2E_FORCE_TESTS to "1" to force the tests to execute.
+//		UNIDOC_ALLOBJECTS_TESTDATA to the path of the corpus folder.
+var (
+	allObjectsCorpusFolder = os.Getenv("UNIDOC_ALLOBJECTS_TESTDATA")
+)
+
+func TestAllObjects(t *testing.T) {
+	if len(allObjectsCorpusFolder) == 0 {
+		if forceTest {
+			t.Fatalf("UNIDOC_ALLOBJECTS_TESTDATA not set")
+		}
+	}
+
+	files, err := ioutil.ReadDir(allObjectsCorpusFolder)
+	if err != nil {
+		if forceTest {
+			t.Fatalf("Error opening %s: %v", allObjectsCorpusFolder, err)
+		}
+		t.Skipf("Skipping allobjects test - unable to open UNIDOC_ALLOBJECTS_TESTDATA (%s)", allObjectsCorpusFolder)
+		return
+	}
+
+	for _, file := range files {
```

Comment:
Maybe files that are not PDFs should be skipped? Right now, the testing stops with an error once a file of another type is encountered.